### PR TITLE
Prom native hashicorp vault

### PR DESCRIPTION
**Pull request by hecomp**
- Added Prom native hashicorp vault

### CODE REVIEW

File: collector/prom-native-vault/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    volumes:
+      - ./consul/config/consul-config.json:/consul/config/config.json
+      - ./consul/data:/consul/data
+
+  consul-worker:
+    container_name: consul-worker
+    restart: always
+    build:
+      context: ./consul
+      dockerfile: Dockerfile
```
Comment (nslaughter) [inline collector/prom-native-vault/docker-compose.yml:53]:
```suggestion
    image: consul
```

**Review by nslaughter — CHANGES_REQUESTED** [review]
Hooray. The example delivers metrics to the environment. Thank you.

Overall looks good on my Mac. 

I requested changes as suggestions, but there are likely a couple of other details you'll need to tweak at the same time to be sure that works. I'm mostly speaking of the change to use the official Consul image. The reason for preferring official images is that they tend to be more secure and robust for cross platform deployment.

File: collector/prom-native-vault/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      VAULT_ADDR: http://0.0.0.0:8200
+    ports:
+      - 8200:8200
+    cap_add:
+      - IPC_LOCK
+    volumes:
+      - ./vault/config:/vault/config
+      - ./vault/config/policies:/vault/config/policies
+      - ./vault/config/data:/vault/data
+      - ./vault/logs:/vault/logs
```
[the remark this answers is not in the document]
Comment (hecomp) [reply collector/prom-native-vault/docker-compose.yml:30]: @nslaughter that would be better :)!! Got it!!

File: collector/prom-native-vault/docker-compose.yml (hunk 1 above)
Comment (hecomp) [reply collector/prom-native-vault/docker-compose.yml]: @nslaughter Done

File: collector/prom-native-vault/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    restart: always
+    build:
+      context: ./consul
+      dockerfile: Dockerfile
+    ports:
+      - 8500:8500
+    command: agent -server -bind 0.0.0.0 -client 0.0.0.0 -bootstrap-expect 1 -config-file=/consul/config/config.json
+    volumes:
+      - ./consul/config/consul-config.json:/consul/config/config.json
+      - ./consul/data:/consul/data
```
[the remark this answers is not in the document]
Comment (hecomp) [reply collector/prom-native-vault/docker-compose.yml]: @nslaughter Done

File: collector/prom-native-vault/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    command: ["server", "-dev", "-config=/vault/config/policies/prometheus-metrics.hcl", "-log-level=info"]
+    depends_on:
+      - consul-server
+
+  consul-server:
+    container_name: consul-server
+    restart: always
+    build:
+      context: ./consul
+      dockerfile: Dockerfile
```
[the remark this answers is not in the document]
Comment (hecomp) [reply collector/prom-native-vault/docker-compose.yml]: @nslaughter Done

File: collector/prom-native-vault/docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hecomp) [reply collector/prom-native-vault/docker-compose.yml]: @nslaughter Done

**Review by nslaughter — APPROVED** [review]

**Pull request merged**